libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -4175,13 +4191,13 @@ string YulUtilFunctions::tryDecodeErrorMessageFunction()
 				let end := add(add(msg, 0x20), length)
 				if gt(end, add(data, sub(returndatasize(), 4))) { leave }
 
-				mstore(<freeMemoryPointer>, add(add(msg, 0x20), <roundUp>(length)))
+				<finalizeAllocation>(data, add(add(offset, msg), add(0x20, length)))
```

Comment:
This looks wrong. `add(offset, msg)` equals `offset + data + offset`. I think that `offset` should be there only once and also `data` should not be passed into the `size` parameter of `finalizeAllocation()`.

Old version was `fmp = data + offset + 0x20 + roundUp(length)`.
The new version is basically `fmp = data + offset + data + offset + 0x20 + length`

```suggestion
				<finalizeAllocation>(data, add(offset, add(0x20, length)))
```

Reply:
Also, was `roundUp()` removed on purpose?

Reply:
roundup is now inside finalizeAllocation

Reply:
If this passed tests then this is apparently not covered. Is something like this testable via inline assembly?

Reply:
It reserved more memory than needed. Maybe when adding gas cost tests, we could also try to check the final value of the free memory pointer somehow?